ssz-rs/benches/compute_proof.rs

```diff
@@ -0,0 +1,56 @@
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use std::hint::black_box;
+
+use ssz_rs::{List, PathElement, Prove};
+use std::convert::TryFrom;
+
+/// Generate fake transactions to benchmark with.
+fn generate_dummy_transactions(size: usize) -> List<[u8; 32], 1000> {
+    let txs = (0..size).map(|i| [i as u8; 32]).collect::<Vec<_>>();
+    List::try_from(txs).expect("Failed to create transaction list")
+}
+
+/// Function to determine transaction indices based on the current size
+/// Returns first, half, and second last indexes.
+fn get_transaction_indices(size: usize) -> Vec<usize> {
+    if size == 0 {
+        vec![]
+    } else {
+        vec![0, size / 2, size - 1]
+    }
```

Comment:
```suggestion
    vec![0, size / 2, size.saturating_sub(1)]
```
It's equivalent code